src/main/java/picard/illumina/parser/IlluminaDataProviderFactory.java

```diff
@@ -256,45 +223,75 @@ public void setApplyEamssFiltering(final boolean applyEamssFiltering) {
         this.applyEamssFiltering = applyEamssFiltering;
     }
 
-    /**
-     * Call this method to create a ClusterData iterator over all clusters for a given tile.
-     *
-     * @param cbcls       A list of cbcls to use when creating this data provider.
-     * @param filterFiles A list of the pf filter files to use when creating this data provider.
-     * @return An iterator for reading the Illumina basecall output for the lane specified in the ctor.
-     */
-    public NewIlluminaDataProvider makeDataProvider(List<File> cbcls,
-                                                    List<AbstractIlluminaPositionFileReader.PositionInfo> locs,
-                                                    File[] filterFiles, int tileNum, File barcodeFile) {
-        return new NewIlluminaDataProvider(cbcls, locs, filterFiles, lane, tileNum, outputMapping, barcodeFile);
+    public BaseIlluminaDataProvider makeDataProvider() {
+        return makeDataProvider((List<Integer>) null,null);
     }
 
-    public BaseIlluminaDataProvider makeDataProvider() {
-        return makeDataProvider(null);
+    public BaseIlluminaDataProvider makeDataProvider(Integer requestedTile) {
+        return makeDataProvider(Collections.singletonList(requestedTile), null);
+    }
+
+    public BaseIlluminaDataProvider makeDataProvider(Integer requestedTile, File barcodeFile) {
+        return makeDataProvider(Collections.singletonList(requestedTile), barcodeFile);
     }
 
     /**
      * Call this method to create a ClusterData iterator over the specified tiles.
      *
      * @return An iterator for reading the Illumina basecall output for the lane specified in the constructor.
      */
-    public BaseIlluminaDataProvider makeDataProvider(List<Integer> requestedTiles) {
+    public BaseIlluminaDataProvider makeDataProvider(List<Integer> requestedTiles, File barcodeFile) {
```

Comment:
This is really weird to me - it takes in a list of tiles and a singular barcode file?  Why is the barcode file even coming from the outside?

Reply:
This is because the new data provider does barcode parsing directly vs deferring to the Parser/Reader. I'll see if I can get around this.